### PR TITLE
Less dependency and more verbose for missing info 

### PR DESCRIPTION
**Pull request by senthil10**
Covered these issues [#58](https://github.com/NationalGenomicsInfrastructure/ngi_reports/issues/58), [#59](https://github.com/NationalGenomicsInfrastructure/ngi_reports/issues/59) and [#60](https://github.com/NationalGenomicsInfrastructure/ngi_reports/issues/60)

### CODE REVIEW

**Comment by ewels** [comment]
Quite a lot to read through here so will go through properly next week. 

One quick thing is that I don't think that `statusdb.py` or `config.py` should be in `ngi_reports/common`. This directory is intended for report content which is common to both Stockholm and Uppsala.

Could you make a new directory `ngi_reports/utils` and put these two scripts there instead please?

I realise that you didn't put `config.py` there in this PR, but still, if you could move it anyway that'd be good :wink:

**Comment by senthil10** [comment]
@ewels : hehe.. no problem, I understand :wink:  and yes, I thought we had a `utils` directory but I couldn't find so put it `common`. I will move those :+1:

**Comment by senthil10** [comment]
@ewels : I will close this `PR` and make new one including [this](https://trello.com/c/tuYOnnqt/468-lookup-x-flowcells-while-generating-report) and some other issues/changes

**Comment by ewels** [comment]
Oops, sorry Senthil - I didn't realise that you'd added more commits since your last comment..!

**Comment by senthil10** [comment]
hehe.. no problem :smile:

**Comment by ewels** [comment]
This PR looks ready to merge - is it not better to merge now and keep the next PR smaller?

File: ngi_reports/stockholm/project_summary.py [hunk 1]
```diff
@@ -413,3 +433,51 @@ def to_ascii(self,value):
...
+        :param config: A config parser instance returned after loading the config file
+        """
+        try:
+            read_length = self.project_info['seq_setup'].split('x')[-1]
+        except:
+            self.LOG.warn("Some problem with fetching setup from db, using default 80% as thershold. But kindly check the sequencing methods in generated report")
+            return default
+        ## Its rare a project run on both hiseq and miseq, if it happened HiSeq threshold will be considered
+        if self.proj_has_hiseq:
+            q_section = "quality_thershold_hiseq"
```
Comment (ewels) [inline ngi_reports/stockholm/project_summary.py]: Typo: `quality_thershold_hiseq` -> `quality_threshold_hiseq`

File: ngi_reports/stockholm/project_summary.py [hunk 2]
```diff
@@ -413,3 +433,51 @@ def to_ascii(self,value):
...
+            return default
+        ## Log warning if the section missing in config file and use default
+        if not config.has_section(q_section):
+            self.LOG.warn("Couldn't find section {} in the config file, using default 80% as threshold".format(q_section))
+            return default
+        ## Get approprite threshold based on runtype
+        try:
+            return int(config.get(q_section, read_length))
+        except NoOptionError:
+            self.LOG.warn("Could not find pre-defined thershold for length {} in section {} in config file, using default 80% as threshold".format(read_length, q_section))
```
Comment (ewels) [inline ngi_reports/stockholm/project_summary.py]: `s/thershold/threshold/` ;)

File: scripts/ngi_reports [hunk 3]
```diff
@@ -168,6 +168,7 @@ if __name__ == "__main__":
     parser.add_argument('-p', '--project', default=None, action="store", help="Project name to generate 'project_summary' report")
     parser.add_argument('-s', '--signature', default=None, action="store", help="Signature/Name for person who generates 'project_summary' report")
     parser.add_argument('-u', '--uppmax_id', default=None, action="store", help="Given UPPMAX id will be used while generating report")
+    parser.add_argument('-q', '--quality', default=None, action="store", type=int, help="Q30(%) thershold for samples to set status")
```
Comment (ewels) [inline scripts/ngi_reports]: `thershold`

**Comment by senthil10** [comment]
@ewels : That works with me as well, it would be easier to review :+1:  

P.S: I will do a `s/thershold/threshold/g` in my next **PR** :joy:

**Comment by ewels** [comment]
I [fixed it for you](https://github.com/NationalGenomicsInfrastructure/ngi_reports/commit/f28e3fe0d507de6ee4154a551c9f77b45a353fdb) during the merge :smile:

**Comment by senthil10** [comment]
hehe.. Tack tack :smile:

**Pull request merged**